File: producer/producer.py
from kafka import KafkaProducer
import time

# Create a Kafka producer
producer = KafkaProducer(bootstrap_servers='kafka:9092')

topic = 'example-topic'

# Send messages with a key to ensure they go to the same partition
for i in range(1, 11):
    key = b'order_key'  # Key to ensure all messages go to the same partition
    value = f'Message {i}'.encode('utf-8')
    producer.send(topic, key=key, value=value)
    print(f"Sent: {value.decode('utf-8')}")
    time.sleep(1)  # Simulate some delay between messages

producer.flush()
producer.close()

File: producer.go
package ICCBD_Project

import (
	"fmt"
	"log"
	"time"

	"github.com/confluentinc/confluent-kafka-go/kafka"
)

func main() {
	producer, err := kafka.NewProducer(&kafka.ConfigMap{
		"bootstrap.servers": "kafka:9092",
		"acks":              "all", // Ensure all brokers confirm message delivery
	})
	if err != nil {
		log.Fatal("Failed to create producer:", err)
	}
	defer producer.Close()

	topic := "test-topic"

	for i := 0; i < 10; i++ {
		message := fmt.Sprintf("Message %d", i)
		err := producer.Produce(&kafka.Message{
			TopicPartition: kafka.TopicPartition{Topic: &topic, Partition: kafka.PartitionAny},
			Value:          []byte(message),
		}, nil)

		if err != nil {
			log.Println("Failed to produce message:", err)
		}
		time.Sleep(1 * time.Second) // Add delay between messages
	}

	producer.Flush(15000) // Wait for message delivery
}
